src/util/codemirror-utils.js

```diff
@@ -0,0 +1,135 @@
+const ignoredTokens = [
+  'suite',
+  'file_input',
+  'simple_stmt',
+  'trailed_atom',
+  'trailer',
+  'comparison', //Dont ignore after boxes are added
+  'testlist_comp',
+]
+
+const tokenColors = {
+  'for_stmt' :    '#F7ABAB',
+  'if_stmt':      '#FFFF00',
+  'number':       'DEA5A4',
+  'exptr_stmt':   '#F7ABD4',
+  'expr':         '#F0ABF7',
+  'str':          '#CAABF7',
+  'atom':         '#ABDBF7',
+  'expr_stmt':    '#BEDA0B',
+  'arglist':      '#F7CDAB',
+  'argument':     '#EAF7AB',
+  'integer':      '#ABF7C6',
+  'funcdef':      '#AEC6CF',
+  'parameters':   '#FFB347',
+  'classdef':     '#03C03C',
+  'try_stmt':     '#966FD6',
+  'pass_stmt':    '#FDFD96',
+  'except_clause':'#779ECB',
+  'while_stmt':   '#F0E68C',
+  'and_expr':     '#F0E68C',
+  'arith_expr':   '#FFA500',
+  'comp_op':      '#DAA520',
+  'dictorsetmaker':'#00FF7F',
+}
+
+function getColor(type) {
+  return tokenColors[type];
+}
+
+/*
+Given a CodeMirror instance, highlight() will use the specified AST and filters
+objects to apply highlighting to the code in the CodeMirror editor.
+*/
+export function highlight(codeMirror, AST, filters) {
+  highlightNode(codeMirror, AST, filters, 'transparent');
+}
+
+// Recursive function for highlighting code in a CodeMirror. highlight() is an
+// exported wrapper func for this, and starts the recursion.
+function highlightNode(codeMirror, node, filters, parentColor) {
+  let color = parentColor;
+
+  // If we aren't ignoring this token...
+  if (ignoredTokens.indexOf(node.type) === -1) {
+    color = getColor(node.type); // Get the color for this token's type
+
+    // If this token has no color 
+    if (!color) {
+      color = 'inherit';
+      console.warn(`token "${node.type}" has no color specified!`);
+    }
+
+    // If this token's filter is not selected
+    if (!filters[node.type] || !filters[node.type].selected) {
+      color = parentColor;
+    }
+
+    // Apply the background color CSS to this token
+    styleRegion(
+      codeMirror,
+      node.begin,
+      node.end,
+      `background-color: ${color};`
+    );
+  }
+
+  // Highlight all children of this token
+  node.children.forEach(child => {
```

Comment:
Avoiding recursion is going to help us with performance. If we were to use something like lodash's [_.forIn](https://lodash.com/docs/4.17.4#forIn), that would simplify the logic of what we're doing, and avoid recursion